packages/flutter_test/lib/src/matchers.dart

```diff
@@ -876,6 +876,15 @@ AsyncMatcher doesNotMeetGuideline(AccessibilityGuideline guideline) {
   return _DoesNotMatchAccessibilityGuideline(guideline);
 }
 
+/// Asserts that an Iterable's elements are in the order defined by `matchers`.
+///
+/// By default, extra items are allowed before, in between and after the items
+/// that should match `matchers`. If `strict` is set to `true`, then no extra
+/// items will be allowed.
+Matcher inOrder(List<Matcher> matchers, {bool strict = false}){
```

Comment:
I wasn't aware of the `containsAllInOrder` matcher, TBH. Even then, when looking at the docs, I had assumed that `containsAllInOrder` wouldn't work with matchers, since all of the doc's examples only use values, and not matchers.

That said, looking into it more, it looks like the `containsAllInOrder` and `orderedEquals` matchers would cover the same use cases this new `inOrder` matcher does. I'll try it out and remove the new matcher if I can.

Reply:
Thanks for pointing this out! I've removed the `inOrder` matcher and pointed to the `containsAllInOrder` and `orderedEquals` methods in the docs for the traversal to make them easier to find for this use case.